scipy/ndimage/tests/test_filters.py

```diff
@@ -2016,6 +2016,38 @@ def test_rank15(self, dtype, xp):
                                      origin=[-1, 0])
         assert_array_almost_equal(expected, output)
 
+    def test_rank16(self):
+        array = [3, 2, 5, 1, 4]
+        expected = np.asarray([3, 3, 2, 4, 4])
+        output = ndimage.rank_filter(array, -2, size=3)
+        assert_array_almost_equal(expected, output)
+
+    def test_rank17(self):
+        array = np.array([3, 2, 5, 1, 4])
+        array.flags.writeable = False
+        expected = np.asarray([3, 3, 2, 4, 4])
+        output = ndimage.rank_filter(array, -2, size=3)
+        assert_array_almost_equal(expected, output)
+
+    def test_rank18(self):
+        array = np.array([3, 2, 5, 1, 4])
+        tested_dtypes = [np.int8, np.int16, np.int32, np.int64, np.float32, np.float64,
+                         np.float16, np.uint8, np.uint16, np.uint32, np.uint64]
```

Comment:
The pattern now used elsewhere in ndimage is to provide dtypes as strings:

```
dt = 'int8'
dtype = getattr(xp, dt)
```

Reply:
done